src/CMakeLists.txt

```diff
@@ -1,10 +1,13 @@
-set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE CACHE BOOL "")
 
 add_subdirectory(common)
 
-list( APPEND CMAKE_INSTALL_RPATH
-  ${CMAKE_INSTALL_PREFIX}/lib
-  ${CMAKE_INSTALL_PREFIX}/lib/hipSYCL/llvm-to-backend)
+if(CMAKE_INSTALL_RPATH_USE_LINK_PATH)
+  list( APPEND CMAKE_INSTALL_RPATH
+    ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}
+    ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/hipSYCL/llvm-to-backend)
```

Comment:
Based on the documentation you have linked, I think most (all?) of our libraries fall under "internal libraries" where rpath is allowed. E.g. I'm fairly certain that disabling rpath unconditionally would probably break binaries using the generic SSCP compiler, as the `llvm-to-backend` libraries would no longer be found.

Reply:
so can we list which are not internal libraries ?
if yes we should put them into a specific dir `%{_libdir}/adaptivecpp/` and update rpath accordingly

Reply:
We do not have a single library that is meant to be directly linked against by users.

The only library we have that outside programs are linked against is our runtime library, `libacpp-rt`. Even in that case the linkage step and rpath-setup is handled internally by our compiler, `acpp`. It's not related at all to the cmake rpath settings.
Because our compiler and runtime library need to be matched exactly (the runtime library does *not* have a stable ABI), I don't think there's a very compelling use case at the moment for not rpath-ing it.